homeassistant/components/refoss/device.py

```diff
@@ -0,0 +1,73 @@
+"""Entity for Refoss."""
+
+from __future__ import annotations
+
+from refoss_ha.controller.device import BaseDevice
+from refoss_ha.enums import Namespace
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+
+class RefossEntity(Entity):
+    """RefossDevice."""
+
+    def __init__(self, device: BaseDevice, channel: int) -> None:
+        """__init__."""
+        self._attr_unique_id = f"refoss.{device.uuid} {channel}"
+        self._attr_name = f"{device.dev_name}"
+        if channel > 0:
+            self._attr_name = f"{device.dev_name}-{channel}"
+        self.device = device
+
+    @property
+    def should_poll(self) -> bool:
+        """should_poll."""
+        return True
+
+    @property
+    def available(self) -> bool:
+        """Return if the device is available."""
+        return True
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """device_info."""
+        device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.device.uuid)},
+            manufacturer="refoss",
+            model=self.device.device_type,
+            name=self.device.dev_name,
+            sw_version=self.device.fmware_version,
+            hw_version=self.device.hdware_version,
+        )
```

Comment:
This is only read when the entity is added. You can set it as `self._attr_device_info = DeviceInfo(....` in `__init__`